app/controllers/courses/accredited_body_controller.rb

```diff
@@ -11,8 +11,12 @@ def edit
 
     def continue
       other_selected_with_no_autocompleted_code = course_params[:accredited_body_code] == "other" && @autocompleted_provider_code.blank?
+      @errors = { accredited_body: ["Please enter at least three letters of the name or provider code"] } if @accredited_body.length < 3
+      @errors = { accredited_body: ["Please enter the name or provider code"] } if @accredited_body.empty?
```

Comment:
Can we move these into methods that help guide us through what we're doing here? Also it'd be nice to not rely on ordering of these validations as this implementation does.

```suggestion
      validate_accredited_body_name_length
      validate_accredited_body_name_presence
```

And these methods might look like:

```ruby
    def validate_accredited_body_name_length
      @errors[accredited_body] ||= []
      if @accredited_body.length > 0 && @accredited_body.length < 3
        @errors[accredited_body] << "Please enter at least three letters of the name or provider code"
      end
    end
```